go/pserver/service.go

```diff
@@ -240,56 +260,46 @@ func (s *Service) GetParam(name string, parameter *Parameter) error {
 	return nil
 }
 
-// pserver save checkpoint
-func (s *Service) doCheckpoint() (err error) {
-	<-s.initialized
-	s.mu.Lock()
-	defer s.mu.Unlock()
+func traceTime(start time.Time, name string) {
+	elapsed := time.Since(start)
+	log.Infof("%s took %v", name, elapsed)
+}
+
+// checkpoint saves checkpoint to disk.
+//
+// checkpoint should be only called after the parameters are
+// initialized.
+func (s *Service) checkpoint() (err error) {
+	log.Infoln("Begin save checkpoint.")
+	defer traceTime(time.Now(), "save checkpoint")
```

Comment:
An idea comes to my mind. We need to put checkpoint log information near the evaluation log.  For example.
```
date=xx, pass = xx, time cost = xx, auc = xxx;
date=xx, pass = xx, time cost = xx, auc = xxx;
date=xx, pass = xx, time cost = xx, auc = xxx;
checkpoint xxx, path=xxx
date=xx, pass = xx, time cost = xx, auc = xxx;
date=xx, pass = xx, time cost = xx, auc = xxx;
```
which we clear know the checkpoint's performance.  This log information is printed in python hook side. 
We can do these things after model evaluation feature done, then send a PR.